src/brogue/Time.c

```diff
@@ -1341,16 +1343,27 @@ void monstersFall() {
                 sprintf(buf2, "%s plunges out of sight!", buf);
                 messageWithColor(buf2, messageColorFromVictim(monst), false);
             }
-            monst->status[STATUS_ENTRANCED] = 0;
-            monst->bookkeepingFlags |= MB_PREPLACED;
-            monst->bookkeepingFlags &= ~(MB_IS_FALLING | MB_SEIZED | MB_SEIZING);
-            monst->targetCorpseLoc[0] = monst->targetCorpseLoc[1] = 0;
+
+            if (rogue.patchVersion < 3) {
+                monst->status[STATUS_ENTRANCED] = 0;
+                monst->bookkeepingFlags |= MB_PREPLACED;
+                monst->bookkeepingFlags &= ~(MB_IS_FALLING | MB_SEIZED | MB_SEIZING);
+                monst->targetCorpseLoc[0] = monst->targetCorpseLoc[1] = 0;
+            }
+
             if (monst->info.flags & MONST_GETS_TURN_ON_ACTIVATION) {
                 // Guardians and mirrored totems never survive the fall. If they did, they might block the level below.
                 killCreature(monst, false);
             } else if (!inflictDamage(NULL, monst, randClumpedRange(6, 12, 2), &red, false)) {
                 demoteMonsterFromLeadership(monst);
 
+                if (rogue.patchVersion >= 3) {
+                    monst->status[STATUS_ENTRANCED] = 0;
+                    monst->bookkeepingFlags |= MB_PREPLACED;
+                    monst->bookkeepingFlags &= ~(MB_IS_FALLING | MB_SEIZED | MB_SEIZING);
+                    monst->targetCorpseLoc[0] = monst->targetCorpseLoc[1] = 0;
```

Comment:
What's the effect of moving this code to this branch (the case where the monster survives the fall)?

Reply:
I just couldn't have the MB_IS_FALLING flag removed before the monster dies. I figured it doesn't make much sense to do all those operations on a potentially dead monster, so I moved all four of those lines after inflictDamage

Reply:
Ok that makes sense, and I think you're right that there couldn't be any unintended effects